routes/invitation.js

```diff
@@ -57,6 +59,15 @@ router.put('/:id/approve', async (req, res) => {
         dateJoined: sender.dateJoined,
       };
 
+      // exit if received is already in user's contacts
+      if (
+        sender.contacts.find(
+          (contact) => contact.email.toString() === receiver.email,
+        )
+      ) {
+        return res.status(400).json({ msg: 'Already in contacts' });
```

Comment:
Should this check be done when the invitation is created rather than when it's approved?